src/Entity/ReactionRuleStorage.php

```diff
@@ -84,6 +85,7 @@ protected function getRegisteredEvents() {
     $events = [];
     foreach ($this->loadMultiple() as $rules_config) {
       $event = $rules_config->getEvent();
+      $event = RulesEventManager::rulesGetEventBaseName($event);
```

Comment:
Not sure what you meant with 'inject'... I implemented as ReactionRuleStorage::eventManager().